### PR TITLE
Improving propagate and filter tests.

### PR DESCRIPTION
**Pull request by drivanov**
## Description

Elimination or suppression of the following warnings that appear in tests implemented in
`common/test_propagate.py` and `common/utils/test_filter.py`
```
================================================================================================= warnings summary =================================================================================================
tests/python/common/utils/test_filter.py: 1 warning
  /usr/local/lib/python3.10/dist-packages/dgl/heterograph.py:92: DGLWarning: Recommend creating graphs by `dgl.graph(data)` instead of `dgl.DGLGraph(data)`.


tests/python/common/test_propagate.py::test_prop_nodes_topo[idtype0]
tests/python/common/test_propagate.py::test_prop_nodes_topo[idtype1]
  /usr/local/lib/python3.10/dist-packages/dgl/core.py:82: DGLWarning: The input graph for the user-defined edge function does not contain valid edges
```

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: tests/python/common/test_propagate.py [hunk 1]
```diff
@@ -100,9 +101,13 @@ def test_prop_nodes_topo(idtype):
     tree.ndata["x"] = F.zeros((5, 2))
     # set all leaf nodes to be ones
     tree.nodes[[1, 3, 4]].data["x"] = F.ones((3, 2))
-    dgl.prop_nodes_topo(
-        tree, message_func=mfunc, reduce_func=rfunc, apply_node_func=None
-    )
+    # Intercepting the warning: The input graph for the user-defined edge
+    # function does not contain valid edges
```
Comment (frozenbugs) [inline tests/python/common/test_propagate.py:105]: Just curious, why not create a graph with valid edges for testing?

Comment (frozenbugs) [inline tests/python/common/test_propagate.py:105]:
```suggestion
    # function does not contain valid edges.
```

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by frozenbugs** [comment]
@dgl-bot

File: tests/python/common/test_propagate.py (hunk 1 above)
Comment (drivanov) [reply tests/python/common/test_propagate.py]: To be honest, I don't understand what's wrong with these edges:
```
    tree.add_nodes(5)
    tree.add_edges(1, 0)
    tree.add_edges(2, 0)
    tree.add_edges(3, 2)
    tree.add_edges(4, 2)
```
however this warning is generated,

Comment (frozenbugs) [reply tests/python/common/test_propagate.py]: If we don't know how the warning is generated, let's avoid suppressing it.
Can you revert this and only submit the change related to tests/python/common/utils/test_filter.py.

Comment (drivanov) [reply tests/python/common/test_propagate.py]: DONE.

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**